sdk/metric/meter.go

```diff
@@ -530,7 +546,10 @@ func (r observer) ObserveFloat64(o metric.Float64Observable, v float64, opts ...
 		return
 	}
 	c := metric.NewObserveConfig(opts)
-	oImpl.observe(v, c.Attributes())
+	measures := r.pipe.float64Measures[oImpl.observableID]
```

Comment:
Do we need to lock around this?  It would be good to ensure our parallel tests Invoke a callback in parallel with RegisterCallbacks

Reply:
Sure, will make the change

Reply:
Pushed the change

Reply:
Tests look good.  Does that mean we don't need to lock around this at all?  Or is there already locking somewhere else?

Reply:
While adding the measure to pipeline we are taking a lock.

Reply:
Looked it up.  We lock around produce(): https://github.com/open-telemetry/opentelemetry-go/blob/06ee6f2be7333e4f53e66fbece967f840e1b0fdc/sdk/metric/pipeline.go#L106-L108

Reply:
Right, read lock is in pipeline already. Sorry, I accidentally mentioned write lock.

Reply:
I spend literally more than half an hour to figure out the same thing. Maybe we could refactor the code to make it cleaner or more readable. Yet, I think it is out of the scope for this PR as I do not think it would be an easy task.
However, can we at least add comments that there is no race condition when accessing `p.float64Measures` and `p.int64Measures` as the pipeline is already locked by `pipeline.Produce`.

Side note: I guess it is possible to achieve race conditions, but these would be in cases of bad usage of the Metrics API.

Reply:
@pellared I completely agree that we should add comments so that anyone reading the code is clear that why there is no lock on that function. I will make the change.

Yes, we can make it more readable by refactoring. I think we should create another issue for the same.

Reply:
I created https://github.com/open-telemetry/opentelemetry-go/issues/5946. You can mention the issue in the comment.

Reply:
I made the change for you https://github.com/open-telemetry/opentelemetry-go/pull/5900/commits/3b96d695b7d43b0d5daa2ebf2ab494833fb79f0e

Reply:
@pellared Thank you so much for this and I really appreciate all feedback, tests look much cleaner now.